src/test/codegen/nontemporal.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -O
+
+#![feature(core_intrinsics)]
+#![crate_type = "lib"]
+
+#[no_mangle]
+pub fn a(a: &mut u64, b: u64) {
+    // CHECK-LABEL: define void @a
+    // CHECK: store i64 %b, i64* %a, align 8, !nontemporal
```

Comment:
The alignment is 4 on i686-musl.

```
[01:00:50] ---- [codegen] codegen/nontemporal.rs stdout ----
[01:00:50] 	
[01:00:50] error: verification with 'FileCheck' failed
[01:00:50] status: exit code: 1
[01:00:50] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/llvm/build/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/nontemporal.ll" "/checkout/src/test/codegen/nontemporal.rs"
[01:00:50] stdout:
[01:00:50] ------------------------------------------
[01:00:50] 
[01:00:50] ------------------------------------------
[01:00:50] stderr:
[01:00:50] ------------------------------------------
[01:00:50] /checkout/src/test/codegen/nontemporal.rs:19:12: error: expected string not found in input
[01:00:50]  // CHECK: store i64 %b, i64* %a, align 8, !nontemporal
[01:00:50]            ^
[01:00:50] /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/nontemporal.ll:7:15: note: scanning from here
[01:00:50] define void @a(i64* nocapture dereferenceable(8) %a, i64 %b) unnamed_addr #0 {
[01:00:50]               ^
[01:00:50] /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/nontemporal.ll:9:2: note: possible intended match here
[01:00:50]  store i64 %b, i64* %a, align 4, !nontemporal !0
[01:00:50]  ^
[01:00:50] 
[01:00:50] ------------------------------------------
[01:00:50] 
[01:00:50] thread '[codegen] codegen/nontemporal.rs' panicked at 'explicit panic', /checkout/src/tools/compiletest/src/runtest.rs:2531:8
[01:00:50] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[01:00:50] 
[01:00:50] 
[01:00:50] failures:
[01:00:50]     [codegen] codegen/nontemporal.rs
[01:00:50] 
[01:00:50] test result: [31mFAILED(B[m. 51 passed; 1 failed; 6 ignored; 0 measured; 0 filtered out
```

Reply:
You can use regexes in codegen tests btw.